scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+                                newValue + ", success=" + answer,
+                       Level.DEBUG);
+        return answer;
+
+    }
+
+    @Override
+    public <T extends Serializable> T replace(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException {
+        T answer = ((Channel<T>) getChannel(channel)).replace(key, value);
+        commitIfNeeded(channel);
+        logWithContext(taskid,
+                       channel,
+                       "Replaced key '" + key + "' with value " + value + ", old value was " + answer,
+                       Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> T computeIfAbsent(String originator, TaskId taskid, String channel, String key,
+            String mappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        T answer;
+        try {
+            Channel<T> chosenChannel = getChannel(channel);
+            answer = chosenChannel.computeIfAbsent(key,
+                                                   (Function<? super String, ? extends T>) evaluateClosure(mappingFunction,
+                                                                                                           Function.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run computeIfAbsent on key '" + key + "', result is: " + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> T computeIfPresent(String originator, TaskId taskid, String channel, String key,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        T answer;
+        try {
+            Channel<T> chosenChannel = getChannel(channel);
+            answer = chosenChannel.computeIfPresent(key,
+                                                    (BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(remappingFunction,
+                                                                                                                         BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run computeIfPresent on key '" + key + "', result is: " + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)